.github/workflows/ci.yml

```diff
@@ -155,3 +155,49 @@ jobs:
           pip install rust_with_cffi/dist/rust_with_cffi*.whl
           python -c "from rust_with_cffi import rust; assert rust.rust_func() == 14"
           python -c "from rust_with_cffi.cffi import lib; assert lib.cffi_func() == 15"
+
+  test-cross-compile:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [
+          { target: "aarch64-unknown-linux-gnu", arch: "aarch64", name: "linux_aarch64" },
+          { target: "armv7-unknown-linux-gnueabihf", arch: "armv7", name: "linux_armv7l" },
+        ]
+    steps:
+      - uses: actions/checkout@master
+      - name: Build Wheels
+        run: |
+          echo 'curl -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain stable
+          source ~/.cargo/env
+          rustup target add ${{ matrix.platform.target }}
+          cd examples/rust_with_cffi/
+          python3.6 -m pip install "crossenv>=1.1.1"
+          python3.6 -m crossenv "/opt/python/cp36-cp36m/bin/python3" --cc $TARGET_CC --cxx $TARGET_CXX --sysroot $TARGET_SYSROOT --env LIBRARY_PATH= --manylinux manylinux1 venv
+          . venv/bin/activate
+          export PYO3_CROSS_LIB_DIR="$PWD/venv/lib"
```

Comment:
I was expecting this to be set by
https://github.com/PyO3/setuptools-rust/blob/503c3390312a904850d3711e5ca9dc685ac2221a/setuptools_rust/build.py#L178

but seems that it didn't.

Reply:
```python
(cross) root@506788c08417:/io/examples/rust_with_cffi# python
Python 3.6.12 (default, Mar 29 2021, 06:48:42)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sysconfig
>>> sysconfig.get_config_var('HOST_GNU_TYPE')
'aarch64-unknown-linux-gnu'
>>> sysconfig.get_config_var('BUILD_GNU_TYPE')
'aarch64-unknown-linux-gnu'
>>> sysconfig.get_path('stdlib')
'/opt/python/cp36-cp36m/lib/python3.6'
>>>
```

Reply:
Huh. It looks like host-python was natively compiled on an aarch64 system. That is something I can fix on my end.

Reply:
Yes, it was copied from manylinux2014_aarch64 docker image: https://github.com/messense/manylinux2014-cross-arm/blob/1da06861a677d9e65e38b51557080e3c505245ff/aarch64.Dockerfile#L133-L134

Reply:
That same issue on my end is also responsible for you needing to specify `--plat-name` to `bdist_wheel` as well.

Reply:
Found another issue, since I've set `CARGO_BUILD_TARGET` it's not going to use `get_nix_target_info()`. 😅

https://github.com/PyO3/setuptools-rust/blob/503c3390312a904850d3711e5ca9dc685ac2221a/setuptools_rust/build.py#L80-L81

Reply:
> That same issue on my end is also responsible for you needing to specify `--plat-name` to `bdist_wheel` as well.

The platform name seems is wrong for armv7l, https://github.com/PyO3/setuptools-rust/pull/141/checks?check_run_id=2242561569

```
ERROR: rust_with_cffi-0.1.0-cp36-abi3-linux_arm.whl is not a supported wheel on this platform.
```

Reply:
I believe that's an artifact of how that version of host-python was compiled. I checked in the docker image, and it was built with `HOST_GNU_TYPE` of `armv7-unknown-linux-gnueabihf`.  The string `armv7l` doesn't appear _anywhere_ in either the compiled binaries or Python source. On a real system, it would get the armv7l from the running operating system.

In this case it seems like the easiest solution would be to pass `--plat-name` as you've been doing, or convince the emulated system your installing it on to accept those wheels. I can also try to add a `--machine` option to crossenv, but it doesn't seem like that actually makes things easier for you: It just changes where you have to add options.

Reply:
I agree. That's why I've pushed https://github.com/messense/manylinux2014-cross-arm/commit/c0e73ddb5e8a4d04f76573993fe095601fee1a2a to force `_PYTHON_HOST_PLATFORM=linux-armv7l`